framework/src/transfers/MultiAppTransfer.C

```diff
@@ -282,10 +293,28 @@ MultiAppTransfer::getAppInfo()
     const auto & ex_from_transform = *_from_transforms[0];
     const auto & ex_to_transform = *_to_transforms[0];
 
+    auto check_transform_compatibility = [this](const MooseCoordTransform & transform)
+    {
+      if (transform.hasNonTranslationTransformation() && !usesMooseCoordTransform())
+        mooseDoOnce(
```

Comment:
I would remove this mooseDoOnce
Since this is shared, it will only output once for the whole simulation, instead of once per transfer type as we had previously